pilot/pkg/bootstrap/server.go

```diff
@@ -1183,7 +1184,64 @@ func (s *Server) initMeshHandlers() {
 	})
 }
 
-func (s *Server) initWorkloadTrustBundle() {
+func (s *Server) addIstioCAToTrustBundle(args *PilotArgs) {
+	// TODO: unify with CreateIstioCA instead of mirroring it
+
+	var err error
+	if s.CA != nil {
+		// If IstioCA is setup, derive trustAnchor directly from CA
+		rootCerts := []string{string(s.CA.GetCAKeyCertBundle().GetRootCertPem())}
+		err = s.workloadTrustBundle.UpdateTrustAnchor(&tb.TrustAnchorUpdate{
+			TrustAnchorConfig: tb.TrustAnchorConfig{Certs: rootCerts},
+			Source:            tb.SourceIstioCA,
+		})
+		if err != nil {
+			log.Errorf("unable to add CA root as trustAnchor")
+		}
+		return
+	}
+
+	// If CA is not running, derive root certificates from the configured CA secrets
```

Comment:
Is the below redundant? I think CA is initiated before running `addIstioCAToTrustBundle`

Reply:
There are scenarios where the that code is never initialized - for instance, when we set ENABLE_CA_SERVER to false. This code is for those scenarios.